File: exec5.c
#include<stdio.h>
#include<stdlib.h>
#include<unistd.h>
#include<string.h>
#include<stdbool.h>

int main()
{
    char fichier[50];
    __pid_t fils;

    while(true)
    {
        printf("Entrez ici le nom du fichier ou entrez q pour quitter : ");
        scanf("%s", fichier);

        if (strcmp(fichier, "q") == 0) break;

        fils = fork();
        if (fils == -1)
        {
            printf("Erreur de création de nouveau processus\n");
            return -1;
        }
        if (fils == 0)
        {
            execl("/usr/bin/emacs", "emacs", fichier, NULL);
        }
    }
    return 0;
}

File: exec5.py
import os

pwd = os.getenv("PWD")

while(True):
    fichier = input("Entrez ici le nom de fichier ou entrez q pour quitter : ")
    if fichier == "q":
        break
    fils = os.fork()
    if fils == 0:
        os.execl("/usr/bin/emacs", "emacs", fichier)